app/controllers/bot_instances_controller.rb

```diff
@@ -2,18 +2,21 @@ class BotInstancesController < ApplicationController
   before_action :authenticate_user!, :only => [:new, :create, :edit, :update, :destroy]
   before_action :set_bot_instance, only: [:show, :edit, :update, :destroy]
   before_action :check_instance_permissions, only: [:edit, :update, :destroy]
-  before_action :check_bot_permissions, only: [:new, :create]
```

Comment:
That works well, thanks!  I still don't understand why the `return` works; if there aren't enough permissions, we want to render the error page and cancel the current request.  After the `return`, isn't Rails just going to continue the request, or does the `render` make it stop?